logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/incremental/RestartSlowSyncProcessForRecoveryUseCase.kt

```diff
@@ -0,0 +1,20 @@
+package com.wire.kalium.logic.sync.incremental
+
+import com.wire.kalium.logic.data.sync.SlowSyncRepository
+
+/**
+ * Restart slowSync process for recovery.
+ */
+interface RestartSlowSyncProcessForRecoveryUseCase {
+    suspend operator fun invoke()
+}
+
+class RestartSlowSyncProcessForRecoveryUseCaseImpl internal constructor(
+    private val slowSyncRepository: SlowSyncRepository
+) : RestartSlowSyncProcessForRecoveryUseCase {
+    override suspend operator fun invoke() {
+        slowSyncRepository.clearLastSlowSyncCompletionInstant()
+        slowSyncRepository.setNeedsToRecoverMLSGroups(true)
+        slowSyncRepository.setNeedsToPersistHistoryLostMessage(true)
```

Comment:
Isn't this kinda by passing the whole 404 recovery logic. 

Can't we set `eventRepository.updateLastProcessedEventId()` to some bogus value and run the incremental sync. That should trigger a real 404 and the recovery logic. Also you don't need add these extra flags.

Reply:
it's only the restarting section, the whole logic is happening the place we call this usecase.
The extra flag is also for the async process to insert the messages after the sync is done.